### ISSUE TITLE
Expose and harden parallel execution for the bisection engine

### ISSUE THREAD

**Issue by ewancarr** [pmsims-package/pmsims#17]
## Summary

The codebase already contains internal parallel support for the bisection engine via `foreach` and `doParallel`, but this is not exposed through the public `simulate_custom()` interface because `parallel = FALSE` is hard-coded in the bisection path.

So this is no longer a generic future idea about parallelisation across the package. The concrete remaining task is to decide whether to support parallel execution in the public API, and if so, to make that path robust and CRAN-friendly.

## Current state

- `calculate_bisection()` has `parallel` and `cores` arguments and can run per-sample-size simulations in parallel.
- `simulate_custom(method = "bisection")` currently forces `parallel = FALSE`.
- The implementation uses `parallel::makeCluster()` plus `foreach` / `doParallel`.
- We should avoid relying on `parallel::detectCores()` as a default strategy without care.

## Scope for this issue

1. Decide whether bisection parallelisation should be user-facing.
2. If yes, expose a public interface for enabling it.
3. Make the backend choice and defaults safe for package use.
4. Add tests and documentation for the parallel path.

## Design questions

- Should the user-facing interface be `parallel` / `cores`, or something higher-level?
- Should we keep the current `foreach` / `doParallel` approach, or move to a different backend such as `future`?
- What should happen on systems where parallel backends are unavailable?
- How do we avoid oversubscription and fragile defaults for worker count?

## Minimum acceptance criteria

- users can intentionally enable parallel bisection runs from the public API
- sequential behavior remains the default
- worker-count behavior is explicit and documented
- tests cover both sequential and parallel code paths
- documentation explains limitations and expected performance tradeoffs

## Out of scope

- full package-wide parallelisation
- parallelising every search engine at once

This should stay focused on the bisection engine unless a broader design is agreed separately.
